actors.md

```diff
@@ -282,33 +282,29 @@ type SlashConsensusFault struct {
 **Algorithm**
 
 ```go
-func shouldSlash(block1, block2 BlockHeader) bool {
+func shouldSlash(block1, block2 BlockHeader, block1Parents TipSet) bool {
 	// First slashing condition, blocks have the same ticket round
 	if sameTicketRound(block1, block2) {
 		return true
 	}
 
-	// Second slashing condition, miner ignored own block when mining
-	// Case A: block2 could have been in block1's parent set but is not
-	block1ParentTipSet := parentOf(block1)
-	if !block1Parent.contains(block2) &&
-		block1ParentTipSet.Height == block2.Height &&
-		block1ParentTipSet.ParentCids == block2.ParentCids {
-		return true
-	}
-
-	// Case B: block1 could have been in block2's parent set but is not
-	block2ParentTipSet := parentOf(block2)
-	if !block2Parent.contains(block1) &&
-		block2ParentTipSet.Height == block1.Height &&
-		block2ParentTipSet.ParentCids == block1.ParentCids {
+	// Second slashing condition, miner ignored own block when mining.
+	// block2 could have been in block1's parent set but is not
+	// verify that block1's parents don't contain block2
+	if block1Parents.Cids == block1.ParentCids &&
+		!block1Parents.contains(block2) &&
+		block1Parents.Height == block2.Height &&
+		block1Parents.ParentCids == block2.ParentCids {
 		return true
```

Comment:
This is to say: Block1Parent is indeed Block1's parent, block1parent omits block 2, even though block 2 should be in block1parent...

Aren't you forgetting a check which is that block2.miner == block1.miner? (otherwise could just be network failure)

you might also check that block2's is the min ticket (rational reason for omitting) but arguably that doesn't matter. Let's penalize either way.